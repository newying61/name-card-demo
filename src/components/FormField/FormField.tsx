import React from 'react';
import Form from 'react-bootstrap/Form';
import { FormInput } from '../Styled/FormInput';

interface FormFieldProps {
  name: string;
  label: string;
  inputType: string;
  placeHolder: string;
  value: string;
  handleChange: any;
  errors: any;
  touched: any;
}

const FormField: React.FC<FormFieldProps> = (
  { name, label, inputType, placeHolder, value, handleChange, errors, touched }) => {
  return (
    <Form.Group controlId={`${name}-field`}>
      <Form.Label>{label}</Form.Label>
      <FormInput
        type={inputType}
        placeholder={placeHolder}
        name={name}
        value={value}
        onChange={handleChange}
        isInvalid={touched[name] && !!errors[name]}
      />
      <Form.Control.Feedback type="invalid">
        {errors[name]}
      </Form.Control.Feedback>
  </Form.Group>
  );
}

export default FormField;