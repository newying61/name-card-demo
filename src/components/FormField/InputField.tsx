import React from 'react';
import Form from 'react-bootstrap/Form';

interface FormFieldProps {
  name: string;
  label: string;
  inputType: string;
  placeHolder: string;
  value: string;
  handleChange: any;
  errors: any;
}

const FormField: React.FC<FormFieldProps> = ({ name, label, inputType, placeHolder, value, handleChange, errors }) => {
  return (
    <Form.Group controlId={`${name}-field`}>
      <Form.Label>{label}</Form.Label>
      <Form.Control
        type={inputType}
        placeholder={placeHolder}
        name={name}
        value={value}
        onChange={handleChange}
        isInvalid={!!errors[name]}
      />
      <Form.Control.Feedback type="invalid">
        {errors[name]}
      </Form.Control.Feedback>
  </Form.Group>
  );
}

export default FormField;