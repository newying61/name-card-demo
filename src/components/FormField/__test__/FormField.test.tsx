import React from 'react';
import { render, cleanup } from '@testing-library/react';
import FormField from '../FormField';

afterEach(cleanup);

describe('FormField', () => {
    it('should render without crash', () => {
      const { container } = render(
        <FormField
          name="test"
          label="test label"
          inputType="text"
          placeHolder="test"
          value="test"
          handleChange={jest.fn()}
          errors={{}}
          touched={{}}
        />
      );

      expect(container.firstChild).toMatchSnapshot();
  });
});