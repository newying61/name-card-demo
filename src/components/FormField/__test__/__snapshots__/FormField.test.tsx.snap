// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField should render without crash 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="test-field"
  >
    test label
  </label>
  <input
    class="sc-bdVaJa btHEdo form-control"
    id="test-field"
    name="test"
    placeholder="test"
    type="text"
    value="test"
  />
  <div
    class="invalid-feedback"
  />
</div>
`;
